src/snowflake/snowpark/dataframe.py

```diff
@@ -1414,6 +1418,11 @@ def select(
                 names.append(e._named())
                 if _emit_ast and _ast_stmt is None:
                     ast_cols.append(e._ast)
+            elif isinstance(e, (list, tuple)):
```

Comment:
Instead of adding the column ASTs manually for this case, please use `build_expr_from_python_val(ast_cols.add(), e)` (from `ast.utils`) to capture when an input provided is a list or tuple.

It would also be very helpful if you could add an expectation test for this in `tests/ast/data/select.test`